test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs

```diff
@@ -392,70 +396,72 @@ public void PushCommand_PushToServerBasicAuth()
-                    return HttpStatusCode.OK;
-                });
-                server.Put.Add("/nuget", r => new Action<HttpListenerResponse>(res =>
+                using (var server = new MockServer())
                 {
-                    var h = r.Headers["Authorization"];
-                    var credential = System.Text.Encoding.Default.GetString(Convert.FromBase64String(h.Substring(6)));
-                    credentialForPutRequest.Add(credential);
-
-                    if (credential.Equals("testuser:testpassword", StringComparison.OrdinalIgnoreCase))
+                    server.Listener.AuthenticationSchemes = AuthenticationSchemes.Basic;
+                    server.Get.Add("/nuget", r =>
                     {
-                        res.StatusCode = (int)HttpStatusCode.OK;
-                    }
-                    else
+                        var h = r.Headers["Authorization"];
+                        var credential = System.Text.Encoding.Default.GetString(Convert.FromBase64String(h.Substring(6)));
+                        credentialForGetRequest.Add(credential);
+                        return HttpStatusCode.OK;
+                    });
+                    server.Put.Add("/nuget", r => new Action<HttpListenerResponse>(res =>
                     {
-                        res.AddHeader("WWW-Authenticate", "Basic ");
-                        res.StatusCode = (int)HttpStatusCode.Unauthorized;
-                    }
-                }));
-                server.Start();
+                        var h = r.Headers["Authorization"];
+                        var credential = System.Text.Encoding.Default.GetString(Convert.FromBase64String(h.Substring(6)));
+                        credentialForPutRequest.Add(credential);
 
-                // Act
-                Environment.SetEnvironmentVariable("FORCE_NUGET_EXE_INTERACTIVE", "true");
-                var args = "push " + packageFileName +
-                    " -Source " + mockServerEndPoint + "nuget";
-                var r1 = CommandRunner.Run(
-                    nugetexe,
-                    packageDirectory,
-                    args,
-                    waitForExit: true,
-                    timeOutInMilliseconds: 10000,
-                    inputAction: (w) =>
-                    {
+                        if (credential.Equals("testuser:testpassword", StringComparison.OrdinalIgnoreCase))
+                        {
+                            res.StatusCode = (int)HttpStatusCode.OK;
+                        }
+                        else
+                        {
+                            res.AddHeader("WWW-Authenticate", "Basic ");
+                            res.StatusCode = (int)HttpStatusCode.Unauthorized;
+                        }
+                    }));
+                    server.Start();
+
+                    // Act
+                    Environment.SetEnvironmentVariable("FORCE_NUGET_EXE_INTERACTIVE", "true");
```

Comment:
this looks like a problem waiting to happen. There is a race between tests here. This should be protected by a mutex.